examples/tutorial/KDD_2025/heteregeneous_walkthrough.ipynb

```
@@ -65,13 +85,557 @@
+    "\n",
+    "validator = kfp_validation_checks(\n",
+    "    job_name=JOB_NAME,\n",
+    "    task_config_uri=TEMPLATE_TASK_CONFIG_PATH,\n",
+    "    resource_config_uri=RESOURCE_CONFIG_PATH,\n",
+    "    start_at=\"config_populator\",\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a465d071",
+   "metadata": {},
+   "source": [
+    "### Config Populator\n",
+    "\n",
+    "Takes in a template `GbmlConfig` and outputs a frozen `GbmlConfig` by populating all job-related metadata paths in `sharedConfig`. These are mostly GCS paths that the following components read and write from, and use as an intermediary data communication medium. For example, the field `sharedConfig.trainedModelMetadata` is populated with a GCS URI, which indicates to the Trainer to write the trained model to this path, and to the Inferencer to read the model from this path. See the full [Config Populator Guide](../../docs/user_guide/overview/components/config_populator.md).\n",
+    "\n",
+    "After running the command below, we will have created a frozen config and uploaded it to the `perm_assets_bucket` provided in the `resource config`. The path to that file will be stored in the file at `FROZEN_TASK_CONFIG_POINTER_FILE_PATH`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "dbebe61a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%capture\n",
+    "# We suppress the output of this cell to avoid cluttering the notebook with logs. Remove %%%capture if you want to see the output.\n",
+    "\n",
+    "!python -m \\\n",
+    "    gigl.src.config_populator.config_populator \\\n",
+    "    --job_name=\"$JOB_NAME\" \\\n",
+    "    --template_uri=\"$TEMPLATE_TASK_CONFIG_PATH\" \\\n",
+    "    --resource_config_uri=\"$RESOURCE_CONFIG_PATH\" \\\n",
+    "    --output_file_path_frozen_gbml_config_uri=\"$FROZEN_TASK_CONFIG_POINTER_FILE_PATH\""
+   ]
```

fyi I'd (personally) prefer to just clear all notebook outputs before submitting and not have this `%%capture` stuff, since when these cells fail the notebook keeps chugging along which is confusing as we'll see downstream failures.

cc: @svij-sc

---

Agree. I added a task to remove all the %%captures. When presenting we can enable scrolling for all cells

---

Honestly agree with this since it makes it so we are aware if things are failing, rather than failing silently -- I'll opt to remove the capture in this case unless folks suggest otherwise.